rules/kuiper.go

```diff
@@ -0,0 +1,306 @@
+	// RuleStatus gets the rule status
+	RuleStatus(name string) (map[string]interface{}, error)
+	// ControlRule starts, stops or restarts the rule
+	ControlRule(name, action string) (*http.Response, error)
+}
+
+type kuiper struct {
+	url        string
+	pluginHost string
+	pluginPort string
+}
+
+var _ KuiperSDK = (*kuiper)(nil)
+
+// NewKuiperSDK instantiates KuiperSDK
+func NewKuiperSDK(url, pluginHost, pluginPort string) KuiperSDK {
+	return &kuiper{
+		url:        url,
+		pluginHost: pluginHost,
+		pluginPort: pluginPort,
+	}
+}
+
+func (k *kuiper) Info() (Info, error) {
+	var i Info
+
+	res, err := http.Get(k.url)
+	if err != nil {
+		return i, errors.Wrap(ErrKuiperServer, err)
+
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&i)
+	if err != nil {
+		return i, errors.Wrap(ErrKuiperServer, err)
+	}
+
+	return i, nil
+}
+
+func (k *kuiper) CreateStream(stream Stream) (*http.Response, error) {
+	sql := k.sql(&stream)
+	body, err := json.Marshal(map[string]string{"sql": sql})
+	if err != nil {
+		return nil, ErrMalformedEntity
+	}
+
+	url := fmt.Sprintf("%s/streams", k.url)
+	res, err := http.Post(url, contentType, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+
+	return res, nil
+}
+
+func (k *kuiper) UpdateStream(stream Stream) (*http.Response, error) {
+	sql := k.sql(&stream)
+	body, err := json.Marshal(map[string]string{"sql": sql})
+	if err != nil {
+		return nil, ErrMalformedEntity
+	}
+
+	url := fmt.Sprintf("%s/streams/%s", k.url, stream.Name)
+	req, err := http.NewRequest("PUT", url, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, ErrMalformedEntity
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+
+	return res, nil
+}
+
+func (k *kuiper) ShowStreams() ([]string, error) {
+	var streams []string
+
+	url := fmt.Sprintf("%s/streams", k.url)
+	res, err := http.Get(url)
+	if err != nil {
+		return streams, errors.Wrap(ErrKuiperServer, err)
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&streams)
+	if err != nil {
+		return streams, errors.Wrap(ErrMalformedEntity, err)
+	}
+	return streams, nil
+}
+
+func (k *kuiper) DescribeStream(name string) (*StreamInfo, error) {
+	url := fmt.Sprintf("%s/streams/%s", k.url, name)
+	res, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	if res.StatusCode == http.StatusNotFound {
+		return nil, errors.Wrap(ErrNotFound, err)
+	}
+	defer res.Body.Close()
+
+	var stream StreamInfo
+	err = json.NewDecoder(res.Body).Decode(&stream)
+	if err != nil {
+		return nil, errors.Wrap(ErrMalformedEntity, err)
+	}
+
+	return &stream, nil
+}
+
+func (k *kuiper) Drop(name, kuiperType string) (*http.Response, error) {
+	if _, ok := KuiperType[kuiperType]; !ok {
+		return nil, ErrMalformedEntity
+	}
+
+	url := fmt.Sprintf("%s/%s/%s", k.url, kuiperType, name)
+	req, err := http.NewRequest("DELETE", url, nil)
```

Comment:
why not `http.Delete`

Reply:
There is no `Delete` in `http` package.

Reply:
You can maybe use `http.MethodDelete`instead of `DELETE` but it's true that there is no `http.Delete`...

Reply:
Actually, `http.MethodDelete` is a string.

Reply:
Right, better const of the package than hardcoded values